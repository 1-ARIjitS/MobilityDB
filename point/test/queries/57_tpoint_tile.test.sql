-------------------------------------------------------------------------------
--
-- This MobilityDB code is provided under The PostgreSQL License.
--
-- Copyright (c) 2016-2021, Université libre de Bruxelles and MobilityDB
-- contributors
--
-- Permission to use, copy, modify, and distribute this software and its
-- documentation for any purpose, without fee, and without a written 
-- agreement is hereby granted, provided that the above copyright notice and
-- this paragraph and the following two paragraphs appear in all copies.
--
-- IN NO EVENT SHALL UNIVERSITE LIBRE DE BRUXELLES BE LIABLE TO ANY PARTY FOR
-- DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING
-- LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION,
-- EVEN IF UNIVERSITE LIBRE DE BRUXELLES HAS BEEN ADVISED OF THE POSSIBILITY 
-- OF SUCH DAMAGE.
--
-- UNIVERSITE LIBRE DE BRUXELLES SPECIFICALLY DISCLAIMS ANY WARRANTIES, 
-- INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
-- AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS ON
-- AN "AS IS" BASIS, AND UNIVERSITE LIBRE DE BRUXELLES HAS NO OBLIGATIONS TO 
-- PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS. 
--
-------------------------------------------------------------------------------

-------------------------------------------------------------------------------
-- Multidimensional tiling
-------------------------------------------------------------------------------

SELECT multidimGrid(tgeompoint '[Point(3 3)@2000-01-15, Point(15 15)@2000-01-25]'::stbox, 2.0) LIMIT 3;
SELECT multidimGrid(tgeompoint 'SRID=3812;[Point(3 3)@2000-01-15, Point(15 15)@2000-01-25]'::stbox, 2.0, geometry 'Point(3 3)') LIMIT 3;
SELECT multidimGrid(tgeompoint '[Point(3 3 3)@2000-01-15, Point(15 15 15)@2000-01-25]'::stbox, 2.0, geometry 'Point(3 3 3)') LIMIT 3;
SELECT multidimGrid(tgeompoint '[Point(3 3)@2000-01-15, Point(15 15)@2000-01-25]'::stbox, 2.0, '2 days', 'Point(3 3)', '2000-01-15') LIMIT 3;
SELECT multidimGrid(tgeompoint '[Point(3 3 3)@2000-01-15, Point(15 15 15)@2000-01-25]'::stbox, 2.0, '2 days', 'Point(3 3 3)', '2000-01-15') LIMIT 3;

SELECT multidimTileStbox(ARRAY[2,0], 2.0);
SELECT multidimTileStbox(ARRAY[2,0,0], 2.0);
SELECT multidimTileStbox(ARRAY[2,0,0], 2.0, interval '2 days');
SELECT multidimTileStbox(ARRAY[2,0,0,0], 2.0, interval '2 days');
SELECT multidimTileStbox(ARRAY[2,0,0,0], 2.0, interval '2 days', geometry 'Point(1 1 1)', '2020-06-15');

-------------------------------------------------------------------------------
