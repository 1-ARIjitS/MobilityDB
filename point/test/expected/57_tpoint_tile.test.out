SELECT multidimGrid(tgeompoint '[Point(3 3)@2000-01-15, Point(15 15)@2000-01-25]'::stbox, 2.0) LIMIT 3;
          multidimgrid          
--------------------------------
 ("{1,1}","STBOX((2,2),(4,4))")
 ("{2,1}","STBOX((4,2),(6,4))")
 ("{3,1}","STBOX((6,2),(8,4))")
(3 rows)

SELECT multidimGrid(tgeompoint 'SRID=3812;[Point(3 3)@2000-01-15, Point(15 15)@2000-01-25]'::stbox, 2.0, geometry 'Point(3 3)') LIMIT 3;
               multidimgrid               
------------------------------------------
 ("{0,0}","SRID=3812;STBOX((3,3),(5,5))")
 ("{1,0}","SRID=3812;STBOX((5,3),(7,5))")
 ("{2,0}","SRID=3812;STBOX((7,3),(9,5))")
(3 rows)

SELECT multidimGrid(tgeompoint '[Point(3 3 3)@2000-01-15, Point(15 15 15)@2000-01-25]'::stbox, 2.0, geometry 'Point(3 3 3)') LIMIT 3;
              multidimgrid              
----------------------------------------
 ("{0,0,0}","STBOX Z((3,3,3),(5,5,5))")
 ("{1,0,0}","STBOX Z((5,3,3),(7,5,5))")
 ("{2,0,0}","STBOX Z((7,3,3),(9,5,5))")
(3 rows)

SELECT multidimGrid(tgeompoint '[Point(3 3)@2000-01-15, Point(15 15)@2000-01-25]'::stbox, 2.0, '2 days', 'Point(3 3)', '2000-01-15') LIMIT 3;
                                   multidimgrid                                   
----------------------------------------------------------------------------------
 ("{0,0,0}","STBOX T((3,3,2000-01-15 00:00:00+00),(5,5,2000-01-17 00:00:00+00))")
 ("{1,0,0}","STBOX T((5,3,2000-01-15 00:00:00+00),(7,5,2000-01-17 00:00:00+00))")
 ("{2,0,0}","STBOX T((7,3,2000-01-15 00:00:00+00),(9,5,2000-01-17 00:00:00+00))")
(3 rows)

SELECT multidimGrid(tgeompoint '[Point(3 3 3)@2000-01-15, Point(15 15 15)@2000-01-25]'::stbox, 2.0, '2 days', 'Point(3 3 3)', '2000-01-15') LIMIT 3;
                                      multidimgrid                                       
-----------------------------------------------------------------------------------------
 ("{0,0,0,0}","STBOX ZT((3,3,3,2000-01-15 00:00:00+00),(5,5,5,2000-01-17 00:00:00+00))")
 ("{1,0,0,0}","STBOX ZT((5,3,3,2000-01-15 00:00:00+00),(7,5,5,2000-01-17 00:00:00+00))")
 ("{2,0,0,0}","STBOX ZT((7,3,3,2000-01-15 00:00:00+00),(9,5,5,2000-01-17 00:00:00+00))")
(3 rows)

SELECT multidimTileStbox(ARRAY[2,0], 2.0);
 multidimtilestbox  
--------------------
 STBOX((4,0),(6,2))
(1 row)

SELECT multidimTileStbox(ARRAY[2,0,0], 2.0);
    multidimtilestbox     
--------------------------
 STBOX Z((4,0,0),(6,2,2))
(1 row)

SELECT multidimTileStbox(ARRAY[2,0,0], 2.0, interval '2 days');
                         multidimtilestbox                          
--------------------------------------------------------------------
 STBOX T((4,0,2000-01-03 00:00:00+00),(6,2,2000-01-05 00:00:00+00))
(1 row)

SELECT multidimTileStbox(ARRAY[2,0,0,0], 2.0, interval '2 days');
                            multidimtilestbox                            
-------------------------------------------------------------------------
 STBOX ZT((4,0,0,2000-01-03 00:00:00+00),(6,2,2,2000-01-05 00:00:00+00))
(1 row)

SELECT multidimTileStbox(ARRAY[2,0,0,0], 2.0, interval '2 days', geometry 'Point(1 1 1)', '2020-06-15');
                            multidimtilestbox                            
-------------------------------------------------------------------------
 STBOX ZT((5,1,1,2020-06-15 00:00:00+00),(7,3,3,2020-06-17 00:00:00+00))
(1 row)

